dspace-api/src/main/java/org/dspace/content/authority/EPersonAuthority.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.content.authority;
+
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.dspace.core.Context;
+import org.dspace.eperson.EPerson;
+import org.dspace.eperson.factory.EPersonServiceFactory;
+import org.dspace.eperson.service.EPersonService;
+import org.dspace.util.UUIDUtils;
+
+/**
+ * Implementation of {@link ChoiceAuthority} based on EPerson. Allows you to set
+ * the id of an eperson as authority.
+ *
+ * @author Mykhaylo Boychuk (4science.it)
+ */
+public class EPersonAuthority implements ChoiceAuthority {
+
+    private static final Logger log = LogManager.getLogger(EPersonAuthority.class);
+
+    /**
+     * the name assigned to the specific instance by the PluginService, @see
+     * {@link NameAwarePlugin}
+     **/
+    private String authorityName;
+
+    private EPersonService ePersonService = EPersonServiceFactory.getInstance().getEPersonService();
+
+    @Override
+    public Choices getBestMatch(String text, String locale) {
+        return getMatches(text, 0, 2, locale);
+    }
+
+    @Override
+    public Choices getMatches(String text, int start, int limit, String locale) {
+        Context context = null;
+        if (limit <= 0) {
+            limit = 20;
+        }
+        context = new Context();
+        List<EPerson> ePersons = null;
+        try {
+            ePersons = ePersonService.search(context, text, start, limit);
```

Comment:
How will this endpoint be made available? I hope it's not possible for submitters to get a list of all EPeople with such a customization

Reply:
Thanks, this limits access already
If I understand it correctly, it does allow every community/collection admin to access it if the property is enabled (not just the community/collection admin of the person collection. But I don't think that's a big issue because community/collection admins can also access the eperson accounts when managing the submitters

@tdonohue do you think we need some warning about that, or can we leave it with this change without warnings?

Reply:
I think the proposed permission settings are reasonable. As you said, EPersons can already been seen by Community/Collection Admins. So, this aligns with that prior decision.